swap_meet/vendor.py

```diff
@@ -1,2 +1,78 @@
+from swap_meet.item import Item
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+        
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.remove(their_item)
+            self.add(their_item)
+            other_vendor.add(my_item)
+            return True
+        else:
+            return False
+        
+    def swap_first_item(self, other_vendor):    
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        self.swap_items(other_vendor, my_item = self.inventory[0] , their_item = other_vendor.inventory[0])
+        
+        return True
+    
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_list.append(item)
+        return category_list
+        
+    
+
+    def get_best_by_category(self, category):
+        items = self.get_by_category(category)
+        best_item = None
+        best_condition = -1.0
+        for item in items:
+            if item.condition > best_condition:
+                best_item = item 
+                best_condition = item.condition
+        return best_item
```

Comment:
Nice loop! The time complexity would be the same, but another way we could write this is using the `max` function with a `lambda` function that tells `max` to use the item's `condition` attribute:

```python
return max(items, key=lambda item: item.condition)
```